### ISSUE TITLE
Big Sur and Apple Silicon Status

### ISSUE THREAD

**Issue by nickv2002** [MaddTheSane/Simple-Comic#41]
TLDR: The App Store version of **SimpleComic won't work in Big Sur on launch day** (today) but should be updated for compatibility soon. Apple Silicon support TBD but expected not to be a problem.

---------

I've been in a multi-step move over the past few months, so haven't had time for side projects but that's closing up now.

For the Mac App store version, I'll look into putting together the newest builds that support Big Sur and Apple Silicon soon.

Looks like Big Sur status was fixed up here: https://github.com/MaddTheSane/Simple-Comic/issues/37

Not sure about Apple Silicon, hopefully it's an easy cross-compile target. I'm getting one of the new MacBooks next week and will test it out then.

**Comment by MaddTheSane** [comment]
I was able to get Apple Silicon to compile fine. But I haven't tested it yet as I don't have an Apple Silicon Mac yet.

**Comment by Myhrdin** [comment]
When do you think the App Store version will be updated ? (Sorry I'm kind of noob on Git and I don't know how to modify Simple comics to apply the fix). Thank you for your work :)

**Comment by nickv2002** [comment]
I'll have some time off next week for the holiday and hope to get to the updates then @Myhrdin 
Thanks for checking in!

**Comment by nickv2002** [comment]
A Big Sur and Apple Silicon native build has been submitted to the App Store for review (version **1.9.0**).
It will be released automatically once Apple approve it there.

**Comment by nickv2002** [comment]
@MaddTheSane note that the app store only accepts 64-bit ARM/x86 builds so I had to change some of the build settings on the `WebPMac` target to remove other architecture refs. It seems these changes are ignored by Git but I figured worth documenting here for my future ref. Also something that might be changed upstream?
![ARM and x86_64 settngs needed for App Store](https://user-images.githubusercontent.com/704392/99887921-43d9ce80-2bfd-11eb-95d3-599ccf9ad819.png)

**Comment by MaddTheSane** [comment]
I was trying to be clever with adding x86_64h architecture. Clearing the _Architectures_ flag should result in the same changes.

**Comment by nickv2002** [comment]
> I was trying to be clever with adding x86_64h architecture. Clearing the _Architectures_ flag should result in the same changes.

Yeah that x86_64h results in a disallowed architecture (`none`) error even though it's 64-bit 🤷‍♂️:
```
ERROR ITMS-90240: "Unsupported Architectures. Your executable contained the following disallowed architectures: '[none (in com.ToWatchList.SimpleComic.pkg/Payload/Simple Comic.app/Contents/Frameworks/WebPMac.framework/Versions/A/WebPMac)]'. New apps submitted to the Mac App Store must support 64-bit starting January 2018, and Mac app updates and existing apps must support 64-bit starting June 2018."
```

**Comment by nickv2002** [comment]
Haven't looked at the code yet but the latest build got rejected for using private APIs. @MaddTheSane if you know what this use is for please advise on removing or reworking it.

> Your app uses or references the following non-public APIs:
> 
> Contents/Library/QuickLook/QuickComic.qlgenerator/Contents/MacOS/QuickComic/_kQLThumbnailPropertyIconFlavorKey
> 
> The use of non-public APIs is not permitted on the App Store because it can lead to a poor user experience should these APIs change. 
> 
> Next Steps
> 
> If you are using third-party libraries, please update to the most recent version of those libraries. If you do not have access to the libraries' source, you may be able to search the compiled binary using the "strings" or "otool" command line tools. The "strings" tool can output a list of the methods that the library calls and "otool -ov" will output the Objective-C class structures and their defined methods. These tools can help you narrow down where the problematic code resides. You could also use the "nm" tool to verify if any third-party libraries are calling these APIs.
>

**Comment by MaddTheSane** [comment]
Okay, _now_ they have a problem with that?
Yeah, I know where it is. I’ll look into possible replacement/removal.

**Comment by MaddTheSane** [comment]
Should be fixed by commit 3b8af881781d0843bac1c2e909e3dbfa8b8b3cb3.

**Comment by nickv2002** [comment]
Submitted updated build to Apple. Thanks for quick turn-around

**Comment by nickv2002** [comment]
Another rejection:

> We discovered one or more bugs in your app when reviewed on Mac running macOS 11.0.1.
> Specifically, your app does not display any contents upon launch. 

This is exactly how Preview.app works... sigh.
I told them it's not a bug and sent them some sample public domain files to try to open.
To help with this in the future, if the app is opened without any associated file, maybe good to show the Open dialog box?

**Comment by nickv2002** [comment]
Updated version is up on the App Store now
https://apps.apple.com/us/app/simple-comic/id1497435571
Please open a new issue if you see any problems. Thanks!
